pandas/_libs/groupby_helper.pxi.in

```diff
@@ -653,37 +643,44 @@ def group_max_{{name}}(ndarray[{{dest_type2}}, ndim=2] out,
                 val = values[i, j]
 
                 # not nan
-                {{if name == 'int64'}}
-                if val != {{nan_val}}:
-                {{else}}
-                if val == val and val != {{nan_val}}:
-                {{endif}}
-                    nobs[lab, j] += 1
-                    if val > maxx[lab, j]:
-                        maxx[lab, j] = val
+                if groupby_t is int64_t:
+                    if val != nan_val:
+                        nobs[lab, j] += 1
+                        if val > maxx[lab, j]:
+                            maxx[lab, j] = val
+                else:
+                    if val == val and val != nan_val:
+                        nobs[lab, j] += 1
```

Comment:
we should have a function that does the null comparisons though with the template code it is slightly 
tricky, maybe have a suite of isna_int, isna_float functions

Reply:
> we should have a function that does the null comparisons though with the template code 

I've been thinking about something similar, will give it a go in the next pass.

Reply:
k